### PR TITLE
Development

### PR DESCRIPTION
**Pull request by VasylShevchenko**

### CODE REVIEW

File: lib/generators/ez/status/install_generator.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module Ez::Status
+  class InstallGenerator < Rails::Generators::Base
+    def create_initializer_file
+      create_file 'config/initializers/ez_status.rb',
+                  "require 'ez/status/providers/database'
+require 'ez/status/providers/cache'
+require 'ez/status/providers/delayed_job'
```
Comment (sveredyuk) [inline lib/generators/ez/status/install_generator.rb:9]: should be commented by default

File: lib/ez/status/all.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+%w(
```
Comment (sveredyuk) [inline lib/ez/status/all.rb:1]: i don't think we need this file

**Review by sveredyuk — COMMENTED** [review]
+ specs required

File: app/controllers/ez/status/status_controller.rb [hunk 3]
```diff
@@ -1,9 +1,26 @@
...
+    class StatusController < ApplicationController
+      before_action :authenticate_with_basic_auth
+
+      def index
+        view 'index'
+      end
+
+      private
+
+      def authenticate_with_basic_auth
```
Comment (sveredyuk) [inline app/controllers/ez/status/status_controller.rb:16]: + add to README

File: lib/ez/status/all.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+[
```
Comment (sveredyuk) [inline lib/ez/status/all.rb:3]: remove this file

File: lib/ez/status/providers/database.rb [hunk 5]
```diff
@@ -4,14 +4,12 @@ module Ez
...
-            ActiveRecord::Base.establish_connection # Establishes connection
-            ActiveRecord::Base.connection # Calls connection object
-            ActiveRecord::Base.connected? ? true : false
-          rescue
-            false
-          end
+        def check
+          ActiveRecord::Base.establish_connection # Establishes connection
+          ActiveRecord::Base.connection # Calls connection object
+          ActiveRecord::Base.connected?
```
Comment (sveredyuk) [inline lib/ez/status/providers/database.rb:10]: [true, 'OK']

File: lib/ez/status/providers/database.rb [hunk 6]
```diff
@@ -4,14 +4,12 @@ module Ez
...
+        rescue
```
Comment (sveredyuk) [inline lib/ez/status/providers/database.rb:11]: rescue StandardError => ex
  [false, ex.message]

File: lib/generators/ez/status/install_generator.rb (hunk 1 above)
Comment (VasylShevchenko) [reply lib/generators/ez/status/install_generator.rb:11]: done

File: lib/ez/status/all.rb (hunk 4 above)
Comment (VasylShevchenko) [reply lib/ez/status/all.rb:3]: done

File: app/controllers/ez/status/status_controller.rb (hunk 3 above)
Comment (VasylShevchenko) [reply app/controllers/ez/status/status_controller.rb:16]: done

File: lib/ez/status/providers/database.rb (hunk 6 above)
Comment (VasylShevchenko) [reply lib/ez/status/providers/database.rb:11]: done

File: lib/ez/status/providers/database.rb (hunk 5 above)
Comment (VasylShevchenko) [reply lib/ez/status/providers/database.rb:10]: done

File: app/cells/ez/status/index_cell.rb [hunk 7]
```diff
@@ -8,8 +8,17 @@ def header
       end
 
       def monitors
-        @monitors ||= Ez::Status.config.monitors.each_with_object({}) do |monitor, acum|
-          acum[monitor.name.demodulize] = monitor.new.check! ? 'OK' : 'FAILURE'
+        @monitors ||= Ez::Status.config.monitors.each_with_object([]) do |monitor, acum|
+          is_working, msg, value = monitor.new.check
```
Comment (sveredyuk) [inline app/cells/ez/status/index_cell.rb:12]: result

File: spec/features/show_spec.rb [hunk 8]
```diff
@@ -62,7 +100,49 @@
...
   end
 
-  describe 'database check' do
+  describe 'check Cache provider' do
+    around do |spec|
+      Ez::Status.configure do |config|
+        config.monitors = [ Ez::Status::Providers::Cache ]
+      end
+      spec.run
+      Ez::Status.configure { |config| config.monitors = [] }
```
Comment (sveredyuk) [inline spec/features/show_spec.rb:109]: Ez::Status.configuration.monitoris =

File: app/cells/ez/status/index_cell.rb (hunk 7 above)
Comment (VasylShevchenko) [reply app/cells/ez/status/index_cell.rb]: done

**Pull request merged**